pkg/metadata_config/metadata_config.go

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2023 SentinelOne, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package metadata_config
+
+import (
+	"fmt"
+	"time"
+)
+
+const (
+	MetadataInterval = time.Minute
+	MinimalInterval  = time.Second
+)
+
+// GetMetadataFunc is the equivalent of Factory.CreateTraces.
+type GetMetadataFunc func() (map[string]string, error)
+
+type DataSetMetadataSettings struct {
```

Comment:
I dont see the point of this class. Why do we need group settings? Why dont we set attributes directly in DataSetClient using builder methods? Do we expect to collector pass attributes which are not know? I assume we wanna log OS version, otel collector version maybe couple of others. But I dont think we wanna give possibility to pass huge map and later on construct huge UserAgent header from it.